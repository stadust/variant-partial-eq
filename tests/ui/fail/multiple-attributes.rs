use variant_partial_eq::VariantPartialEq;

#[derive(Debug, VariantPartialEq)]
struct MyStruct<'a> {
    #[variant_compare = "nonsense"]
    #[variant_compare = "nonsense2"]
    my_str: &'a str
}

fn main() {
    let s = "Hello World!".to_owned();

    assert_eq!(MyStruct {my_str: &s}, MyStruct {my_str: "Hello World!"});
}