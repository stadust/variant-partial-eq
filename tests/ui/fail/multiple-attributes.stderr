error: Expected only single #[variant_compare = "..."] attribute
 --> tests/ui/fail/multiple-attributes.rs:6:7
  |
6 |     #[variant_compare = "nonsense2"]
  |       ^^^^^^^^^^^^^^^
